### PR TITLE
Walkshed generation

### PR DESCRIPTION
**Pull request by allthesignals**
This Pull Request takes care of a few things:

1. Generates .5, 1, 1.5, and 2-mile isochrones along sidewalk and bikelane networks
2. Associates newly submitted survey responses with their intersecting walkshed
3. Adds new explicit associations across models
4. Complicated queries are moved into query objects

@mzagaja if you would kindly review the models and their hook actions - there is some fundamental rails stuff there and I'm curious if you might anticipate any issues with how I've implemented. Thanks!

### CODE REVIEW

File: app/queries/intersection_query.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class IntersectionQuery
```
Comment (mzagaja) [inline app/queries/intersection_query.rb]: This appears to be a [lazy class](https://sourcemaking.com/refactoring/smells/lazy-class) and can probably be combined with SurveyResponse.

File: app/queries/walkshed_query.rb [hunk 2]
```diff
@@ -0,0 +1,120 @@
+class WalkshedQuery
```
Comment (mzagaja) [inline app/queries/walkshed_query.rb]: This also appears to be a lazy class and can probably be combined with its calling class.

File: app/models/school.rb [hunk 3]
```diff
@@ -13,65 +15,31 @@ def has_active_survey?
...
+      shed_20_ring = shed_20.difference(shed_15).to_s
+
+      # updates with new rings
+      update_columns({  shed_10: shed_10_ring,
+                        shed_15: shed_15_ring,
+                        shed_20: shed_20_ring  })
+      
+    end
+
+    def now
```
Comment (mzagaja) [inline app/models/school.rb]: This method does not appear to be used? It's probably ok to explicitly call DateTime.now without encapsulating in a method anyways.

**Comment by mzagaja** [comment]
Sorry for sort of going beyond the scope of what you asked for, I didn't see the comment about extracting the queries before I suggested combining them. I think I'd still suggest not breaking them out for now unless they're being re-used, I think it's ok for them to live in their model. But you can use your own judgement if you plan to recycle them often and/or expand them.

Also you should probably [add .DS_Store to your global .gitignore](http://stackoverflow.com/questions/18393498/gitignore-all-the-ds-store-files-in-every-folder-and-subfolder).

Otherwise I don't see any glaring issues as far as the functions go. So overall looks good to me, and you can follow the thoughts you think make sense.

**Comment by allthesignals** [comment]
Thanks @mzagaja, it's probably less confusing to move the query objects back into their calling classes. I'll work on that before merging. Thanks again for reviewing.

**Pull request merged**